src/components/swap/SwapModalFooter.tsx

```diff
@@ -1,36 +1,100 @@
 import { Trans } from '@lingui/macro'
-import { Trade } from '@uniswap/router-sdk'
-import { Currency, TradeType } from '@uniswap/sdk-core'
+import { Currency, Percent, TradeType } from '@uniswap/sdk-core'
+import { ElementName, EventName } from 'components/AmplitudeAnalytics/constants'
+import { Event } from 'components/AmplitudeAnalytics/constants'
+import { TraceEvent } from 'components/AmplitudeAnalytics/TraceEvent'
+import { useStablecoinValue } from 'hooks/useStablecoinPrice'
+import useTransactionDeadline from 'hooks/useTransactionDeadline'
 import { ReactNode } from 'react'
 import { Text } from 'rebass'
+import { InterfaceTrade } from 'state/routing/types'
+import { useUserSlippageTolerance } from 'state/user/hooks'
 
 import { ButtonError } from '../Button'
 import { AutoRow } from '../Row'
 import { SwapCallbackError } from './styleds'
 
+const formatAnalyticsEventProperties = (
+  trade: InterfaceTrade<Currency, Currency, TradeType>,
+  txHash: string | undefined,
+  allowedSlippage: Percent,
+  transactionDeadlineSecondsSinceEpoch: number | undefined,
+  isAutoSlippage: boolean,
+  tokenInAmountUsd: string | undefined,
+  tokenOutAmountUsd: string | undefined
+) => ({
+  estimated_network_fee_usd: trade.gasUseEstimateUSD ? parseFloat(trade.gasUseEstimateUSD?.toFixed(2)) : undefined,
+  transaction_hash: txHash,
+  transaction_deadline_seconds: transactionDeadlineSecondsSinceEpoch
+    ? /** durationInSeconds= */ transactionDeadlineSecondsSinceEpoch - new Date().getTime() / 1000
```

Comment:
what does this comment mean?

Reply:
I'm trying to indicate that it's calculating the duration (in seconds) till the transaction deadline, by subtracting current time (expressed in seconds since epoch) from the transaction deadline (in seconds since epoch) in the future. do you think a different comment would make this clearer?

Reply:
yeah i would say that in a sentence rather than a variable

Reply:
created a helper function with better naming to reduce confusion: getDurationTillTimestampSinceEpoch